src/main/java/com/restaurant/service/impl/UserServiceImpl.java

```diff
@@ -33,13 +33,13 @@ public UserDto createUser(UserDto userDto) {
 
 		User user1 = userRepo.findByEmail(userDto.getEmail());// validate by pattern matcher
 
-		if (user1 == null) {
+		if (user1 == null && Keywords.EMAIL_REGEX.matcher(userDto.getEmail()).matches()) {
```

Comment:
check regex match before finding by email in the database.

Reply:
Done